internal/diagnostics/diagnostics.go

```diff
@@ -54,3 +54,15 @@ func (m *Message) Format(args ...any) string {
 	}
 	return text
 }
+
+func FormatMessage(m *Message, args ...any) *Message {
+	return &Message{
```

Comment:
Probably could be a `m.Clone()` followed by a write to `text`, right?

Reply:
I don't think we have a `Clone` method on `diagnostics.Message`. But I could just be an assignment copy followed by a write to the `text` property.

Reply:
Ah, right, I was mistaking this for the type you just added `Clone` to above.